File: solutions/fortran/reverse-string/1/reverse_string.f90
module reverse_string
   implicit none
contains

   function reverse(input) result(reversed)
      character(*), intent(in) :: input
      character(len=len(input)) :: reversed
      integer :: i
      do i = 1, len(input)
         reversed(i:i) = input(len(input) - i + 1:len(input) - i + 1)
      end do
   end function

end module

File: solutions/cpp/reverse-string/1/reverse_string.cpp
#include "reverse_string.h"

namespace reverse_string
{
    std::string reverse_string(std::string s)
    {
        std::string result(s.size(), ' ');

        for (size_t i = 0; i < s.size(); i++)
        {
            result[i] = s[s.size() - i - 1];
        }

        return result;
    }
}
